package/kedro_viz/data_access/repositories/modular_pipelines.py

```diff
@@ -210,6 +210,10 @@ def extract_from_node(self, node: GraphNode) -> Optional[str]:
 
         modular_pipeline = self.get_or_create_modular_pipeline(modular_pipeline_id)
 
+        # Inherit tags from the nodes of the modular pipeline.
+        if modular_pipeline_id:

```

Comment:
on line 208, we are already checking if modular_pipeline_id exists, I feel this check is redundant.